pkg/services/ngalert/remote/remote_primary_forked_alertmanager.go

```diff
@@ -37,19 +37,26 @@ func (fam *RemotePrimaryForkedAlertmanager) GetStatus() apimodels.GettableStatus
 }
 
 func (fam *RemotePrimaryForkedAlertmanager) CreateSilence(ctx context.Context, silence *apimodels.PostableSilence) (string, error) {
-	return "", nil
+	_, err := fam.internal.CreateSilence(ctx, silence)
+	if err != nil {
+		return "", err
+	}
+	return fam.remote.CreateSilence(ctx, silence)
 }
 
 func (fam *RemotePrimaryForkedAlertmanager) DeleteSilence(ctx context.Context, id string) error {
-	return nil
+	if err := fam.remote.DeleteSilence(ctx, id); err != nil {
+		return err
+	}
+	return fam.internal.DeleteSilence(ctx, id)
```

Comment:
Does this work? I didn't think we kept the `id` of the silence from the internal AM, and the `id` is a UUID, so will be different on each AM?

Reply:
There are two fixes that we could look at:

1. Instead of forwarding the HTTP request to the Mimir AM, we send it the protobuf silence (either via the bucket or over HTTP/gRPC). This will allow the silences to be consistent.

2. We keep a table that maps Silence IDs in the internal AM with their respective silence in the remote AM. A silence is not created or deleted until both sides of the table are present/absent. We can keep this table in the kvstore in the database.